### PR TITLE
Add @disposableComputed

### PR DESCRIPTION
**Pull request by BrendanAnnable**
More memory leaks to disappear!

### CODE REVIEW

File: src/client/base/disposable_computed.ts [hunk 1]
```diff
@@ -2,12 +2,31 @@ import { IComputedValue } from 'mobx'
...
 import { computed } from 'mobx'
 
+function disposableComputedDecorator(target: any, propertyKey: string, descriptor: PropertyDescriptor) {
+  if (!descriptor) {
+    descriptor = Object.getOwnPropertyDescriptor(target, propertyKey)!
+  }
+  const originalMethod = descriptor.get!
+  // Lazily create and store the computed expression per instance.
+  // This allows us to bind each computed expression to the correct `this` context.
+  const expressionPerInstance = new WeakMap()
```
Comment (TrentHouliston) [inline src/client/base/disposable_computed.ts]: Pfft, someone should do this the proper way someday

File: src/client/components/three/stories/three.stories.tsx [hunk 2]
```diff
@@ -139,7 +140,7 @@ class BoxViewModel {
     return mesh
   }
 
-  @computed
+  @disposableComputed
   private get material(): Material {
     return new MeshPhongMaterial({ color: this.model.color })
```
Comment (TrentHouliston) [inline src/client/components/three/stories/three.stories.tsx]: Does this still recompile the shader every time you change the colour?

Comment (BrendanAnnable) [reply src/client/components/three/stories/three.stories.tsx]: Yes 😭

Comment (TrentHouliston) [reply src/client/components/three/stories/three.stories.tsx]: And once again all of your wonderful abstractions must die in service of performance :P

Comment (BrendanAnnable) [reply src/client/components/three/stories/three.stories.tsx]: JUST DON'T CHANGE COLOURS OKAY EVERYONE 😭

Comment (BrendanAnnable) [reply src/client/components/three/stories/three.stories.tsx]: JUST DON'T DO ANYTHING USEFUL 😭

Comment (BrendanAnnable) [reply src/client/components/three/stories/three.stories.tsx]: I think we can solve this, but it might be tricky

**Pull request merged**